lnrpc/routerrpc/router_server.go

```diff
@@ -1506,6 +1514,119 @@ func (s *Server) HtlcInterceptor(stream Router_HtlcInterceptorServer) error {
 	).run()
 }
 
+// XAddLocalChanAliases is an experimental API that creates a set of new
+// channel SCID alias mappings. The list of successfully created mappings is
+// returned. This is only a locally stored alias, and will not be communicated
+// to the channel peer via any message. Therefore, routing over such an alias
+// will only work of the peer also calls this same RPC on their end.
+func (s *Server) XAddLocalChanAliases(_ context.Context,
+	in *AddAliasesRequest) (*AddAliasesResponse, error) {
+
+	// Create a map that will store the successfully created scid mappings.
+	scidMap := make(
+		map[lnwire.ShortChannelID][]lnwire.ShortChannelID,
+	)
+
+	for _, v := range in.AliasMaps {
+		baseScid := lnwire.NewShortChanIDFromInt(v.BaseScid)
+
+		existingAliases := s.cfg.AliasMgr.GetAliases(baseScid)
+
+		for _, alias := range v.Aliases {
+			exists := false
+
+			// Check if the alias already exists in the alias map.
+			for _, existingAlias := range existingAliases {
```

Comment:
should check against all existing aliases rather than only for this `baseScid`

Reply:
Good point, added.